core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/executionrequest/ExecutionRequestRepositoryDtoTest.java

```diff
@@ -397,6 +396,97 @@ void getAllFilteredByLabel() {
                 );
     }
 
+    @Test
+    @WithIesiUser(username = "spring",
+            authorities = {
+                    "EXECUTION_REQUESTS_READ@PUBLIC"
+            })
+    void getAllFilteredByRunId() {
+        LocalDateTime requestTimestamp = LocalDateTime.now();
+        Map<String, Object> executionRequest1Map = ExecutionRequestBuilder.generateExecutionRequest(1, requestTimestamp, 2, 1, "script1", 1L, "PUBLIC", "test", 1, 1);
+
+        executionRequestConfiguration.insert((ExecutionRequest) executionRequest1Map.get("executionRequest"));
+        scriptConfiguration.insert((Script) executionRequest1Map.get("script"));
+
+        UUID runId = UUID.randomUUID();
+
+        ScriptExecution scriptExecution = ScriptExecution.builder()
+                .scriptExecutionKey(new ScriptExecutionKey(UUID.randomUUID().toString()))
+                .scriptRunStatus(ScriptRunStatus.RUNNING)
+                .startTimestamp(LocalDateTime.now())
+                .endTimestamp(LocalDateTime.now().plus(1L, ChronoUnit.MILLIS))
+                .scriptExecutionRequestKey(new ScriptExecutionRequestKey(executionRequest1Map.get("scriptExecutionRequest10UUID").toString()))
+                .runId(runId.toString())
+                .build();
+        scriptExecutionConfiguration.insert(scriptExecution);
+
+        ExecutionRequestDto executionRequestDto = (ExecutionRequestDto) executionRequest1Map.get("executionRequestDto");
+        executionRequestDto.getScriptExecutionRequests().iterator().next().setRunId(runId.toString());
+        executionRequestDto.getScriptExecutionRequests().iterator().next().setRunStatus(ScriptRunStatus.RUNNING);
+
+        assertThat(executionRequestDtoRepository.getAll(SecurityContextHolder.getContext().getAuthentication(), PageRequest.of(0, 1),
+                Stream.of(new ExecutionRequestFilter(ExecutionRequestFilterOption.RUN_ID, runId.toString(), false)).collect(Collectors.toList())))
+                .containsOnly(
+                        (ExecutionRequestDto) executionRequest1Map.get("executionRequestDto")
+                );
+    }
+
+    @Test
+    @WithIesiUser(username = "spring",
+            authorities = {
+                    "EXECUTION_REQUESTS_READ@PUBLIC"
+            })
+    void getAllFilteredByRunIdMultiple() {
```

Comment:
script executions cannot have the same run id. You can hardcode two run-ids that have some part in common and then filter on the common part